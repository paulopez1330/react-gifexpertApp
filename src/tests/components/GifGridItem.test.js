import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('Pruebas en <GifGridItem/>', () => {

    const title = 'mi titulo';
    const url   = 'https://localhost/image.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('debe de mostrar componente correctamente', () => {
                
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe tener un parrafo con el texto', () => {
        
        const p = wrapper.find('p');

        expect( p.text().trim () ).toBe(title);        
    })

    test('debe tener una imagen al url y alt de los props', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('debe tenerel fadein', () => {
        
        const d = wrapper.find('div');
        const className = d.prop('className');
        
        expect( className.includes('animate__fadeIn') ).toBe( true );
    })    
})
