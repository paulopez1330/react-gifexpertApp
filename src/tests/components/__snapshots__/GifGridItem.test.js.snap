// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar componente correctamente 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="mi titulo"
    src="https://localhost/image.jpg"
  />
  <p>
     
    mi titulo
     
  </p>
</div>
`;
